### PR TITLE
Story/file monitoring/story3 controller and endpoint

### PR DESCRIPTION
**Pull request by ChristianMorup**

### CODE REVIEW

File: RTDS/RTDS/Configuration/ConfigurationManager.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                {
+                    Logger.Fatal("Base source path is missing.");
+                    throw new MissingSpecifiedPathException("Base source path is missing.");
+                }
+            }
+
+            return _configuration.Paths;
+        }
+
+        public static void OverrideConfiguration(RTDSConfiguration configuration, bool overrideConfigFile)
```
Comment (Cpilgaard) [inline RTDS/RTDS/Configuration/ConfigurationManager.cs]: Why this override method? Dont quite understand it

File: RTDS/RTDS/Configuration/Exceptions/MissingSpecifiedPathException.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        public MissingSpecifiedPathException(string path)
+        {
+            MissingPath = path;
+        }
+
+        public MissingSpecifiedPathException(string path, Exception e)
+        {
+            _e = e;
+            MissingPath = path;
+        }
```
Comment (Cpilgaard) [inline RTDS/RTDS/Configuration/Exceptions/MissingSpecifiedPathException.cs]: Why 3 MissingSpecifiedPathException?

File: RTDS/RTDS/Configuration/Exceptions/NoConfigurationFileException.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        {
+            Path = path;
+        }
+
+        public NoConfigurationFileException(string path, Exception e)
+        {
+            _e = e;
+            Path = path;
+        }
+
```
Comment (Cpilgaard) [inline RTDS/RTDS/Configuration/Exceptions/NoConfigurationFileException.cs]: Same as above comment

File: RTDS/RTDS/Configuration/Data/RTDSPaths.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿namespace RTDS.Configuration.Data
+{
+    public class RTDSPaths
+    {
+        public string BaseTargetPath { get; set; }
+
```
Comment (MikkelHarbo) [inline RTDS/RTDS/Configuration/Data/RTDSPaths.cs]: Spacing

File: RTDS/RTDS/Monitoring/Args/SearchDirectoryArgs.cs [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
     {
-        public SearchDirectoryArgs(string path, string name)
+        public SearchDirectoryArgs(string path, string name, IMonitor relatedMonitor)
         {
             Path = path;
             Name = name;
+            RelatedMonitor = relatedMonitor;
         }
 
         public string Path { get; }
```
Comment (MikkelHarbo) [inline RTDS/RTDS/Monitoring/Args/SearchDirectoryArgs.cs]: spacing

File: RTDS/RTDS/Monitoring/Args/SearchDirectoryArgs.cs [hunk 6]
```diff
@@ -1,17 +1,21 @@
...
 
         public string Name { get; }
+
```
Comment (MikkelHarbo) [inline RTDS/RTDS/Monitoring/Args/SearchDirectoryArgs.cs]: spacing

File: RTDS/RTDS/Monitoring/Factory/IMonitorFactory.cs [hunk 7]
```diff
@@ -1,7 +1,11 @@
-﻿namespace RTDS.Monitoring.Factory
+﻿using RTDS.Monitoring.Monitors;
+
+namespace RTDS.Monitoring.Factory
 {
     internal interface IMonitorFactory
     {
         IFileMonitor CreateFileMonitor();
+
```
Comment (MikkelHarbo) [inline RTDS/RTDS/Monitoring/Factory/IMonitorFactory.cs]: spacing

File: RTDS/RTDS/Utility/FolderCreator.cs [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+        private string CreateMhaFolderPath(string basePath)
+        {
+            return Path.Combine(basePath, CreateMhaFolderName());
+        }
+
+        private string CreateMhaFolderName()
+        {
+            return "mha";
+        }
```
Comment (Cpilgaard) [inline RTDS/RTDS/Utility/FolderCreator.cs]: Do we need logging here?

File: RTDS/RTDS/Monitoring/Monitors/FolderMonitor.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+namespace RTDS.Monitoring.Monitors
+{
+    internal class FolderMonitor : AbstractMonitor, IMonitor
+    {
+        public override event EventHandler<SearchDirectoryArgs> Created;
+        private static readonly NLog.Logger Logger = NLog.LogManager.GetCurrentClassLogger();
+
+        public FolderMonitor(IFileSystemWatcherWrapper watcher) : base(watcher)
+        {
+        }
```
Comment (MikkelHarbo) [inline RTDS/RTDS/Monitoring/Monitors/FolderMonitor.cs]: Purpose?

File: RTDS/RTDS/Monitoring/Monitors/IMonitor.cs [hunk 10]
```diff
@@ -2,13 +2,15 @@
...
 {
-    internal interface  IMonitor
+    internal interface IMonitor
     {
         Task StartMonitoringAsync(string path);
 
-        string MonitoredPath {get; }
+        string MonitoredPath { get; }
+
+        Guid Guid { get; }
```
Comment (MikkelHarbo) [inline RTDS/RTDS/Monitoring/Monitors/IMonitor.cs]: Spacing

File: RTDS/RTDS/Utility/FolderCreator.cs [hunk 11]
```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+using RTDS.DTO;
+using ConfigurationManager = RTDS.Configuration.ConfigurationManager;
+
+namespace RTDS.Utility
+{
+    internal class FolderCreator : IFolderCreator
```
Comment (MikkelHarbo) [inline RTDS/RTDS/Utility/FolderCreator.cs]: Why is it internal?

File: RTDS/RTDS/Utility/IFileUtil.cs [hunk 12]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Threading.Tasks;
+
+namespace RTDS.Utility
+{
+    internal interface IFileUtil
+    {
+        Task<string> CopyFileAsync(string sourceFile, string destinationFile);
+
```
Comment (MikkelHarbo) [inline RTDS/RTDS/Utility/IFileUtil.cs]: Spacing

File: RTDS/RTDS/Utility/IFolderCreator.cs [hunk 13]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading.Tasks;
+using RTDS.DTO;
+
+namespace RTDS.Utility
+{
+    internal interface IFolderCreator
+    {
+        Task<ProjectionFolderStructure> CreateFolderStructureForProjectionsAsync();
+
```
Comment (MikkelHarbo) [inline RTDS/RTDS/Utility/IFolderCreator.cs]: Spacing

**Pull request merged**